Composer/packages/client/src/CreationFlow/DefineConversation/index.js

```diff
@@ -11,46 +13,77 @@ import { TextField } from 'office-ui-fabric-react/lib/TextField';
 import { LocationSelectContent } from '../LocationBrowser/LocationSelectContent';
 import { styles as wizardStyles } from '../StepWizard/styles';
 
+import { StoreContext } from './../../store';
 import { name, description } from './styles';
 
-const nameRegex = /^[a-zA-Z0-9-_.]+$/;
-
-const validateForm = data => {
-  const errors = {};
-  const { name } = data;
-
-  if (!name || !nameRegex.test(name)) {
-    errors.name = formatMessage(
-      'Spaces and special characters are not allowed. Use letters, numbers, -, or _., numbers, -, and _'
-    );
-  }
-
-  return errors;
-};
+const MAXTRYTIMES = 10000;
 
 export function DefineConversation(props) {
-  const { onSubmit, onGetErrorMessage, onDismiss, enableLocationBrowse } = props;
-
-  const [formData, setFormData] = useState({ errors: {} });
+  const {
+    onSubmit,
+    onDismiss,
+    enableLocationBrowse,
+    focusedStorageFolder,
+    onCurrentPathUpdate,
+    currentPath,
+    bots,
+  } = props;
+
+  const { state } = useContext(StoreContext);
+  const { templateId, botName } = state;
+  const [formData, setFormData] = useState({ name: getDefaultName() });
+  const [formDataErrors, setFormDataErrors] = useState({ errors: {} });
```

Comment:
`{ errors: {} }` is a different type than what `setFormDataErrors` is being invoked with. You should convert this file to typescript to catch this kind of thing.